phac-nml/irida-next
# Data Table Component and update to profiles PAT table

## What does this PR do and why?
This PR adds the `DataTableComponent` which acts as the shell for our table components and will eventually be implemented into all our tables.

To start, the `PersonalAccessTokens` table has been updated to use the `DataTableComponent`

## Screenshots or screen recordings
![image](https://github.com/user-attachments/assets/8f50f348-39da-4821-9f3c-02f6fb36d1cd)

## How to set up and validate locally
1. Navigate to the access tokens page via User Settings > Access Tokens
2. Verify the access tokens table lists the proper data as expected
3. Verify `Revoke` works as expected
4. Additional user feedback (flash messages) has been added to this page. Test the following and ensure the proper messages are displayed:
- Create access token successfully
- Create access token failure with no scopes selected
- Revoke token success
5. Verify empty table state when no personal access tokens

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
